astropy/units/quantity_helper/function_helpers.py

```diff
@@ -252,15 +251,63 @@ def like_helper(a, *args, **kwargs):
     return (a.view(np.ndarray),) + args, kwargs, unit, None
 
 
+def _nanop_helper(a, *, out, _out_unit_from_a):
+    from astropy.units import Quantity
+
+    if out is not None:
+        if not isinstance(out, Quantity):
+            # TODO: for an ndarray output, we could in principle
+            # try converting all Quantity to dimensionless.
+            raise NotImplementedError
+        out_unit = out.unit
```

Comment:
So I think this check goes to the detailed code, just using `getattr(a, "unit", u.one)`